code/modules/reagents/reagent_containers/glass.dm

```diff
@@ -84,6 +84,47 @@
 		if(standard_pour_into(user, target))
 			return
 
+		//Help the user avoid clumsy splashing mistakes due to finicky sprites etc.
+		if (reagents.total_volume) //don't pop a dialog if it's empty...
+			//handle lidded containers
+			if  (user.clumsy == 0 && (istype(target, /obj/item/weapon/reagent_containers)))
+				var/resp = alert(user, "Do you really want to splash the solution?", "Splash the solution?", "Abort", "Proceed")
+				if (resp == "Abort")
+					return
+				else if (!user.Adjacent(target))
+					user << "You cannot reach the [src]."
+					return
+				else if (user.stat != 0)
+					return
+				else if (src.loc != user)
+					user << "You need to hold the [src] to splash with it."
+					return
+
+			//handles splashes onto floors and effects (puddles, blood etc.)
+			if (user.clumsy <= 1 && (istype(target, /turf) || istype(target, /obj/effect) ))
+
+				//resolve contentless effects
+				var/intarget = null
+				if (istype(target, /obj/effect))
+					intarget = target.loc
+				else intarget = target
+
+				//iterate over things in content
+				for (var/thing in intarget)
+					if (is_type_in_list(thing, can_be_placed_into))
+						var/response = alert(user, "Do you really want to splash the solution?", "Splash the solution?", "Abort", "Proceed")
```

Comment:
@Iamgoofball The break statement stops the checking once the first occurence has been found.